src/goto-programs/goto_program.cpp

```diff
@@ -166,22 +176,29 @@ std::ostream &goto_programt::output_instruction(
         instruction.code.find(ID_exception_list).get_sub();
       assert(instruction.targets.size()==exception_list.size());
       for(instructiont::targetst::const_iterator
-          gt_it=instruction.targets.begin();
+            gt_it=instruction.targets.begin();
           gt_it!=instruction.targets.end();
           gt_it++,
-          i++)
+            i++)
       {
         if(gt_it!=instruction.targets.begin())
           out << ", ";
         out << exception_list[i].id() << "->"
             << (*gt_it)->target_number;
       }
     }
-    else
+    else if(instruction.code.get_statement()==ID_pop_catch)
+    {
       out << "CATCH-POP";
+    }
+    else
+    {
+      out << "! unexpected CATCH opcode " << instruction.code.get_statement();
```

Comment:
Shouldn't this be `UNREACHABLE`? I don't think this should be printed to the user as is as it is an internal error.